### PR TITLE
#9338 Add Rich Text Editor toolbar buttons

### PR DESCRIPTION
**Pull request by mnholtz**
## What does this PR do?

- Part of #9338
- Adds the following buttons to the Rich Text Editor toolbar:
    - Underline
    - Strikethrough
    - Numbered List
    - Bulleted List
    - Horizontal rule
- [x] TODO: Collapse buttons under ellipsis menu when screen is too narrow
- [x] TODO: Add remaining test cases for horizontal rule and strikethrough

## Discussion

- I'm aware that the button toolbar actions have some obvious repetition that can be extracted. I propose we focus on doing that in a future slice, per sake of time.

## Demo

image

image

## Future Work

- The following buttons need to be added to a follow-up slice:
    - Link button 
        - Url insertion dialog was more complicated than I anticipated. [See this branch here for progress on that](https://github.com/pixiebrix/pixiebrix-extension/compare/main...feature/9338_add_link_toolbar_action)
        - Please ask me for a sync update on this one
    - Blockquote button
        - Blockquote doesn't seem to have any styling applied to it in any of our document renderers. We might considering adding this button to nice-to-have

### CODE REVIEW

**Comment by grahamlangford** [comment]
> Blockquote button
Blockquote doesn't seem to have any styling applied to it in any of our document renderers. We might considering adding this button to nice-to-have

Agreed. Unless @Pashaminkovsky disagrees

**Comment by Pashaminkovsky** [comment]
I agree, blockquote is not a priority.

File: src/components/richTextEditor/RichTextEditor.tsx [hunk 1]
```diff
@@ -26,7 +27,7 @@ const RichTextEditor: React.FunctionComponent<EditorProviderProps> = (
 ) => (
   <div className={styles.root}>
```
Comment (grahamlangford) [inline src/components/richTextEditor/RichTextEditor.tsx:28]: NIT: can't this be applied to the `editorContainerProps` prop of `EditorProvider`?

File: src/components/richTextEditor/toolbar/BulletedListButton.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { useCurrentEditor } from "@tiptap/react";
+import { Button } from "react-bootstrap";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faListUl } from "@fortawesome/free-solid-svg-icons";
+
+const BulletedListButton: React.FunctionComponent = () => {
+  const { editor } = useCurrentEditor();
+
+  if (!editor) {
+    return null;
```
Comment (grahamlangford) [inline src/components/richTextEditor/toolbar/BulletedListButton.tsx:11]: I wonder if we should be asserting an error that `BulletedListButton must be a part of an Editor Context` or something similar? Not a priority now, just something to think about

Comment (grahamlangford) [reply src/components/richTextEditor/toolbar/BulletedListButton.tsx:11]: Assuming there's not another reason `editor` could be null

**Review by grahamlangford — APPROVED** [review]

File: src/components/richTextEditor/RichTextEditor.tsx (hunk 1 above)
Comment (mnholtz) [reply src/components/richTextEditor/RichTextEditor.tsx:28]: `editorContainerProps` only refers to the content editable portion of the editor (per it being a headless library). We are wrapping the entire editor in our own style that also includes the toolbar

File: src/components/richTextEditor/toolbar/BulletedListButton.tsx (hunk 2 above)
Comment (mnholtz) [reply src/components/richTextEditor/toolbar/BulletedListButton.tsx:11]: > Assuming there's not another reason editor could be falsy
I think that assumption is probably right. I'll consider this when I go to refactor the button components

**Pull request merged**